lib/actions/api.js

```diff
@@ -83,86 +82,96 @@ function getActiveItinerary (otpState) {
  */
 export function routingQuery (searchId = null) {
   return async function (dispatch, getState) {
+    // FIXME: batchId is searchId for now.
     const state = getState()
     const otpState = state.otp
 
     const isNewSearch = !searchId
     if (isNewSearch) searchId = randId()
-    const routingType = otpState.currentQuery.routingType
     // Don't permit a routing query if the query is invalid
     if (!queryIsValid(otpState)) {
       console.warn('Query is invalid. Aborting routing query', otpState.currentQuery)
       return
     }
     const activeItinerary = getActiveItinerary(otpState)
-    dispatch(routingRequest({ activeItinerary, routingType, searchId }))
-
-    // fetch a realtime route
-    const query = constructRoutingQuery(otpState)
-    fetch(query, getOtpFetchOptions(state))
-      .then(getJsonAndCheckResponse)
-      .then(json => {
-        dispatch(routingResponse({ response: json, searchId }))
-        // If tracking is enabled, store locations and search after successful
-        // search is completed.
-        if (otpState.user.trackRecent) {
-          const { from, to } = otpState.currentQuery
-          if (!isStoredPlace(from)) {
-            dispatch(rememberPlace({ type: 'recent', location: formatRecentPlace(from) }))
-          }
-          if (!isStoredPlace(to)) {
-            dispatch(rememberPlace({ type: 'recent', location: formatRecentPlace(to) }))
+    const routingType = otpState.currentQuery.routingType
+    // For multiple mode combinations, gather injected params from config/query.
+    // Otherwise, inject nothing (rely on what's in current query) and perform
+    // one iteration.
+    const iterations = otpState.config.modes && otpState.config.modes.combinations
+      ? otpState.config.modes.combinations.map(({mode, params}) => ({mode, ...params}))
+      : [{}]
```

Comment:
Won't this result in no trip plans happening with the current trimet-call-taker config?

Reply:
No, it should result in one, the empty object represents the trip plan -- just with no injected params.